### PR TITLE
Additions to modelzoo APIs

### PR DESCRIPTION
**Pull request by yeshaokai**
Spatial pyramid is applied by default in video adaptation now, to act as filtering and to reduce quantization error

If it's topview, gradient masking is also applied in video adaptation

Some other stuff for developers' use

### CODE REVIEW

**Review by MMathisLab — APPROVED** [review]

**Comment by MMathisLab** [comment]
@yeshaokai please fix tests errors

**Comment by jeylau** [comment]
@yeshaokai, looks good to me. Could we fix #2212 in this one though before merging?

**Review by jeylau — APPROVED** [review]

**Comment by yeshaokai** [comment]
I fixed the empty prediction following the user's advice. After the tests pass we should be good.

File: deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py [hunk 1]
```diff
@@ -63,7 +65,8 @@ def train(
     cfg["pseudo_threshold"] = pseudo_threshold
     cfg["video_path"] = video_path
     cfg["traintime_resize"] = traintime_resize
-
+    cfg["topview"] = topview
```
Comment (MMathisLab) [inline deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py]: can you explain why this is needed?

**Review by MMathisLab — CHANGES_REQUESTED** [review]

File: deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py (hunk 1 above)
Comment (yeshaokai) [reply deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py]: To add gradient masking in video adaptation. If the user choose high threshold for video adaptation, then many keypoints are dropped thus masking is needed. But "masking for video adaptation" is only applicable for topview. During inference time,  masking for  side-view quadruped models will make the model confuse occluded keypoints and visible but filtered keypoints .

Comment (MMathisLab) [reply deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py]: lets minimally change to ["superanimal"] then topview is okay; then we are forward thinking for new models. Agree @yeshaokai then we can merge actually

File: deeplabcut/pose_estimation_tensorflow/predict_supermodel.py [hunk 2]
```diff
@@ -20,6 +20,10 @@ def video_inference_superanimal(
     video_adapt=False,
     plot_trajectories=True,
     pcutoff=0.1,
+    init_weights = '',
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_tensorflow/predict_supermodel.py]: This argument isn't used in the method, it should be removed

File: examples/testscript_superanimal_adaptation.py [hunk 3]
```diff
@@ -41,15 +41,3 @@
...
-
-    superanimal_name = "superanimal_quadruped"
-    deeplabcut.video_inference_superanimal(
-        [video],
-        superanimal_name,
-        videotype=".mp4",
-        video_adapt=True,
-        scale_list=scale_list,
-        pcutoff=0.3,
-    )
```
Comment (n-poulsen) [inline examples/testscript_superanimal_adaptation.py]: As we have `video_adapt=True`, adaptation training will be run for the (default) number of iterations, which is 1000. I think this is a bit slow for a test script,  and we should only run the adaptation for ~50 iterations.

```python
        pcutoff=0.3,
        adapt_iterations=50,
    )
```

**Review by n-poulsen — DISMISSED** [review]

**Review by MMathisLab — APPROVED** [review]

**Comment by yeshaokai** [comment]
@n-poulsen Thanks for the edits! They look fine to me

**Review by MMathisLab — APPROVED** [review]

**Review by AlexEMG — APPROVED** [review]

**Pull request merged**